src/Compilers/CSharp/Portable/SourceGeneration/CSharpGeneratorDriver.cs

```diff
@@ -55,9 +56,15 @@ public static CSharpGeneratorDriver Create(params IIncrementalGenerator[] increm
         /// <param name="additionalTexts">A list of <see cref="AdditionalText"/>s available to generators in this driver, or <c>null</c> if there are none.</param>
         /// <param name="parseOptions">The <see cref="CSharpParseOptions"/> that should be used when parsing generated files, or <c>null</c> to use <see cref="CSharpParseOptions.Default"/></param>
         /// <param name="optionsProvider">An <see cref="AnalyzerConfigOptionsProvider"/> that can be used to retrieve analyzer config values by the generators in this driver, or <c>null</c> if there are none.</param>
+        /// <param name="disabledOutputs">A bit field of <see cref="IncrementalGeneratorOutputKind"/> that indicates the output types that should be diabled and not produce ouput.</param>
         /// <returns>A new <see cref="CSharpGeneratorDriver"/> instance.</returns>
-        public static CSharpGeneratorDriver Create(IEnumerable<ISourceGenerator> generators, IEnumerable<AdditionalText>? additionalTexts = null, CSharpParseOptions? parseOptions = null, AnalyzerConfigOptionsProvider? optionsProvider = null)
-            => new CSharpGeneratorDriver(parseOptions ?? CSharpParseOptions.Default, generators.ToImmutableArray(), optionsProvider ?? CompilerAnalyzerConfigOptionsProvider.Empty, additionalTexts.AsImmutableOrEmpty());
+        public static CSharpGeneratorDriver Create(IEnumerable<ISourceGenerator> generators, IEnumerable<AdditionalText>? additionalTexts = null, CSharpParseOptions? parseOptions = null, AnalyzerConfigOptionsProvider? optionsProvider = null, IncrementalGeneratorOutputKind disabledOutputs = IncrementalGeneratorOutputKind.None)
```

Comment:
>IncrementalGeneratorOutputKind disabledOutputs [](http://example.com/codeflow?start=242&length=46)

Will we have additional generator options in the future? If we know now that we will, perhaps it would make sense to pass an options type here rather than a single value.

Reply:
Yes, we plan to have more, but `IncrementalGeneratorOutputKind` is a `[Flags]` so we can combine them together like `IncrementalGeneratorOutputKind.Implementation | IncrementalGeneratorKind.SomeOtherOutput` to build up which nodes we don't want.

Reply:
Adding more `IncrementalGeneratorOutputKind` values can be accomodated with the current overload but do we expect to add options other than `IncrementatlGeneratorOutputKind`?

Reply:
Oh, I see what you're saying. Yeah actually that probably makes sense.